pageserver/src/repository.rs

```diff
@@ -96,24 +107,26 @@ pub trait Timeline: Send + Sync {
     fn history<'a>(&'a self) -> Result<Box<dyn History + 'a>>;
 }
 
-pub trait History: Iterator<Item = Result<RelationUpdate>> {
+pub trait History: Iterator<Item = Result<Modification>> {
     /// The last_valid_lsn at the time of history() call.
     fn lsn(&self) -> Lsn;
 }
 
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct RelationUpdate {
-    pub rel: RelTag,
+pub struct Modification {
+    pub tag: ObjectTag,
     pub lsn: Lsn,
-    pub update: Update,
+    pub data: Vec<u8>,
 }
```

Comment:
Can we add documentation to this struct, to describe its purpose?

Reply:
This structure replaces RelationUpdate used by branch push, because now we copy not only relation data but also non-relational data. Should I add comments myself (because this code was written by Patric)? And should I do it in this branch?
I really want to first merge this branch with main and only then do refactorings whch are not directly related with Object tags.

Reply:
> This structure replaces RelationUpdate used by branch push, because now we copy not only relation data but also non-relational data. Should I add comments myself (because this code was written by Patric)? And should I do it in this branch?

Would be good to do that, as a separate PR, thanks. It was not clear to me either what a RelationUpdate was, until I browsed around to see where it's used.

Reply:
Code for copying branches has to be changed in any way if we switched to new object model.
Is it really sensible to do this work twice: first in old manner using high level object storage operations and only then using much simple approach by just copying objects at low level (as key-value pairs without interpreting their content)?